exasol/error/_parse.py

```diff
@@ -162,31 +200,55 @@ def _validate_mitigations(
                     for e in invalid
                 ]
             )
-
-    def _validate_parameters(self, node: ast.Dict, file: str) -> None:
-        # Validate parameters
-        for key in node.keys:
-            if not isinstance(key, ast.Constant) and key is not None:
-                self._errors.append(
-                    self.Error(
-                        message=self._error_msg.format(type="key", value=type(key)),
-                        file=file,
-                        line_number=key.lineno,
-                    )
+            if invalid:
+                return None
+            else:
+                return node
+        elif isinstance(node, ast.Constant):
+            return node
+        return None
+
+    def _validate_parameters(self, node: ast.expr, file: str) -> Optional[ast.Dict]:
+        if not isinstance(node, ast.Dict):
+            self._errors.append(
+                self.Error(
+                    message=self._error_msg.format(type="parameters", value=type(node)),
+                    file=file,
+                    line_number=node.lineno,
                 )
-        for value in node.values:
-            if isinstance(value, ast.Call):
-                description = value.args[1]
-                if not isinstance(description, ast.Constant):
+            )
+            return None
+        else:
+            is_ok = True
+            # Validate parameters
+            for key in node.keys:
+                if not isinstance(key, ast.Constant) and key is not None:
                     self._errors.append(
                         self.Error(
-                            message=self._error_msg.format(
-                                type="description", value=type(description)
-                            ),
+                            message=self._error_msg.format(type="key", value=type(key)),
                             file=file,
-                            line_number=value.lineno,
+                            line_number=key.lineno,
                         )
                     )
+                    is_ok = False
+            for value in node.values:
```

Comment:
maybe extracting values validation into its own method is a good idea

Reply:
Done